js/modules/loki_public_chat_api.js

```diff
@@ -39,25 +45,23 @@ class LokiPublicServerAPI {
     this.channels = [];
   }
   findOrCreateChannel(channelId, conversationId) {
-    let thisChannel = null;
-    this.channels.forEach(channel => {
-      if (
-        channel.channelId === channelId &&
-        channel.conversationId === conversationId
-      ) {
-        thisChannel = channel;
-      }
-    });
-    if (thisChannel === null) {
+    let thisChannel = this.channels.find(channel => channel.channelId === channelId);
+    if (!thisChannel) {
       thisChannel = new LokiPublicChannelAPI(this, channelId, conversationId);
       this.channels.push(thisChannel);
     }
     return thisChannel;
   }
   unregisterChannel(channelId) {
-    // find it, remove it
-    // if no channels left, request we deregister server
-    return channelId || this; // this is just to make eslint happy
+    const thisChannel = this.channels.find(channel => channel.channelId === channelId);
```

Comment:
if we converted to a for in or for loop we could get rid of the 2nd search (go o(n) instead of o(n * 2))

Reply:
Good call